src/Storage/Sha1FileHashingService.php

```diff
@@ -179,10 +178,23 @@ private function getCache()
     private function buildCacheKey($fileID, $fs)
     {
         $fsID = $this->getFilesystemKey($fs);
-
         return base64_encode(sprintf('%s://%s', $fsID, $fileID));
     }
 
+    /**
+     * Return the current timestamp for the provided file or the current time if the file doesn't exists.
+     * @param string $fileID
+     * @param string|Filesystem $fs
+     * @return string
+     */
+    private function getTimestamp($fileID, $fs)
+    {
+        $filesystem = $this->getFilesystem($fs);
+        return $filesystem->has($fileID) ?
+            $filesystem->getTimestamp($fileID) :
+            DBDatetime::now()->getTimestamp();
```

Comment:
They are instances where we'll be generating a hash value for a file that doesn't exists yet, so we need a fallback in case the file doesn't exists.

Reply:
Just had a chat with @NightJar about the virtue of return a fake timestamp. We were originally doing this, because we were making it part of the key and we sometime needed to generate hash keys for files that might not exists.

After some consideration, there's a scenario where this is useful. If you're moving a file around, the FysystemAssetStore will try to move the existing hash value to a different hash key. If you decide to move the hash before the physical file, you won't have a file there yet. By generating, a timestamp equal to "now", FlysystemAssetStore won't have to worry about this intricacy. So we reduce complexity a little bit.